will/backends/io_adapters/slack.py

```diff
@@ -457,85 +619,75 @@ def _update_channels(self):
-            if not hasattr(self, "me") or not self.me:
+            if self.me is None:
                 self.me = self.load("slack_me_cache", None)
-            if not hasattr(self, "handle") or not self.handle:
+            if self.handle is None:
                 self.handle = self.load("slack_handle_cache", None)
         else:
             self._people = people
             self.save("slack_people_cache", people)
             self.save("slack_me_cache", self.me)
             self.save("slack_handle_cache", self.handle)
 
-    def _update_backend_metadata(self):
-        self._update_people()
-        self._update_channels()
+    def _update_backend_metadata(self, **event):
+        "Updates all our internal caches.  Very expenseive"
+        logging.debug("updating backend on event: %s", event)
+        name = event.get("data", dict()).get("self", dict()).get("name")
+        if name is not None:
+            self.__class__.handle = name
+        Thread(
+            target=self._update_people, args=(event["web_client"],), daemon=True
+        ).start()
+        Thread(
+            target=self._update_channels, args=(event["web_client"],), daemon=True
+        ).start()
 
     def _watch_slack_rtm(self):
-        while True:
-            try:
-                if self.client.rtm_connect(auto_reconnect=True):
-                    self._update_backend_metadata()
-
-                    num_polls_between_updates = 30 / settings.EVENT_LOOP_INTERVAL  # Every 30 seconds
-                    current_poll_count = 0
-                    while True:
-                        events = self.client.rtm_read()
-                        if len(events) > 0:
-                            # TODO: only handle events that are new.
-                            # print(len(events))
-                            for e in events:
-                                self.handle_incoming_event(e)
-
-                        # Update channels/people/me/etc every 10s or so.
-                        current_poll_count += 1
-                        if current_poll_count > num_polls_between_updates:
-                            self._update_backend_metadata()
-                            current_poll_count = 0
-
-                        self.sleep_for_event_loop()
-            except (WebSocketConnectionClosedException, SlackConnectionError):
-                logging.error('Encountered connection error attempting reconnect in 2 seconds')
-                time.sleep(2)
-            except (KeyboardInterrupt, SystemExit):
-                break
-            except:
-                logging.critical("Error in watching slack RTM: \n%s" % traceback.format_exc())
-                break
+        "This is our main loop."
+        # The decorators don't work on unbound methods.  Sigh.
+        # These are all events that should spark an update of our inventory
+        # RTMClient.run_on(event="hello")(self._update_backend_metadata)
```

Comment:
Can we eliminate these lines or is the intention to execute these updates on other event types in the future?

Reply:
Good question.  Some of these events we absolutely don't care about.  Others, we really should do an incremental update rather than a full update.  Yet at the same time, we probably want to do a full update periodically just to be sure we didn't miss anything.  The docs say that the "hello" event should be the first one you get, and that you should load data on that.  I did not find that to be the case.  So I left in these commented events as hints to myself, but really, they're probably not useful to anyone but myself.